#Project title: Summary of the current satellite situation and distribution



Project definition: Satellites are now reaching a stage of commercialize usage, with 5540 satellite launches and 36 satellite attributes being recorded in the dataset, the project will be focused on discovering the similarities between satellites, finding patterns and determining the satellite trends. Ultimately provide a summary guide to map out the current satellites situation and supply insights about the satellite distribution.


Data source: https://www.ucsusa.org/resources/satellite-database (UCS Satellite Database)

Motivation: As SpaceX reaches success in satellite launch, it shows how satellites have been pushed towards a stage of commercialization. With more investment opportunities for satellites, it is paramount that companies know about the current satellite distribution.Therefore, summary of a satellite distribution can be a good guide to follow.


##Phase 1

### Exploratory 

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
#mount the drive
from google.colab import drive
drive.mount('/content/drive')
data_folder = '/content/Shareddrives/MyDrive/ba780-a4-fall2/'

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
#import the csv file
import pandas as pd

satellite = pd.read_csv('/content/drive/Shareddrives/ba780-a4-fall22/Dataset/UCS-Satellite-Database-5-1-2022.csv')
print(satellite.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5467 entries, 0 to 5466
Data columns (total 68 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Name of Satellite, Alternate Names  5465 non-null   object 
 1   Current Official Name of Satellite  5465 non-null   object 
 2   Country/Org of UN Registry          4968 non-null   object 
 3   Country of Operator/Owner           5465 non-null   object 
 4   Operator/Owner                      5465 non-null   object 
 5   Users                               5465 non-null   object 
 6   Purpose                             5465 non-null   object 
 7   Detailed Purpose                    1140 non-null   object 
 8   Class of Orbit                      5465 non-null   object 
 9   Type of Orbit                       4843 non-null   object 
 10  Longitude of GEO (degrees)          5464 non-null   float64
 11  Perigee (km)                        5464 no

1. In order to define the id of the dataframe for each record, we first want to exam if 'Name of Satellite, Alternate Names' variables are unique, so that can be used as ID for each observation. As the result, number of unique Satellites are: 5458, and there are: 5467 total number of satellite launched, meaning that satellites with same name can be launched twice. So in next step, we will generate the unique identifier for each satellite.

In [ ]:
## Count unique Name of Stallite, Alternate Names, and see if each row is unique
Num_Satellite = len(satellite['Name of Satellite, Alternate Names'].unique())
Total_Record = len(satellite['Name of Satellite, Alternate Names'])

Total_Record > Num_Satellite
"Number of unique Satellites are: {}, and there are: {} total number of satellite launched".format(len(satellite['Name of Satellite, Alternate Names'].unique()),len(satellite['Name of Satellite, Alternate Names']))

'Number of unique Satellites are: 5458, and there are: 5467 total number of satellite launched'

2. Using the for loop to generate the unique identifier and e

In [ ]:
#Dropping all the unamed and source columns
satellite_clean_1 = (satellite.drop(satellite.iloc[:,28:68] , axis = 1)).copy()
satellite_clean_1.info()
#satellite_clean_1.to_csv('sample_data/satellite_version2.csv')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5467 entries, 0 to 5466
Data columns (total 28 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Name of Satellite, Alternate Names  5465 non-null   object 
 1   Current Official Name of Satellite  5465 non-null   object 
 2   Country/Org of UN Registry          4968 non-null   object 
 3   Country of Operator/Owner           5465 non-null   object 
 4   Operator/Owner                      5465 non-null   object 
 5   Users                               5465 non-null   object 
 6   Purpose                             5465 non-null   object 
 7   Detailed Purpose                    1140 non-null   object 
 8   Class of Orbit                      5465 non-null   object 
 9   Type of Orbit                       4843 non-null   object 
 10  Longitude of GEO (degrees)          5464 non-null   float64
 11  Perigee (km)                        5464 no

TypeError: ignored